segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/ReadIndexBlockMetadata.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.metadata;
+
+import io.pravega.common.ObjectBuilder;
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.io.IOException;
+
+/**
+ * Represents read index block .
+ * Following metadata is stored.
+ * <ul>
+ * <li>Name of entry.</li>     
+ * <li>Name of the chunk containing first byte of this block.</li>
+ * <li>Start offset of the chunk.</li>
+ * </ul>
+ */
+@Builder(toBuilder = true)
+@Data
+@EqualsAndHashCode(callSuper = true)
+@ThreadSafe
```

Comment:
How is this thread safe if none of your mutable internal state is protected by any mutexes?